src/jv_parse.c

```diff
@@ -512,6 +512,12 @@ static pfunc check_literal(struct jv_parser* p) {
   switch (p->tokenbuf[0]) {
   case 't': pattern = "true"; plen = 4; v = jv_true(); break;
   case 'f': pattern = "false"; plen = 5; v = jv_false(); break;
+  case '\'':
+    return "Invalid string literal; expected \", but got '";
+  case '/':
+  case '*':
+  case '&':
+    return "Invalid literal";
```

Comment:
Why?

Reply:
In @wader's comment above, he found other character's that were triggering the `invalid numeric literal`. This check catches those. Would this cause an issue?

Reply:
Why should it say "Invalid literal" instead of "Invalid numeric literal" for those three characters? How is that better?

Reply:
Does it make more sense for those to trigger `Invalid numeric literal`? Personally those don't seem like numeric characters (as compared to, say, `-` which if alone, could be an invalid negative number). I was just implementing the suggestions above. I'm happy to revert if this doesn't make sense.

Reply:
"Invalid numeric literal" means that `jq` tried to parse a JSON number and it was not a number. so the error makes sense, yes.
"Invalid literal"? what does that even mean? :)

Reply:
Oh I see. I guess if we got to this point, then we've tried parsing it as a string, boolean, and null. So all that's left is parsing it as a number, hence that message. Maybe the better error message would be something like `Invalid character '&' encountered while parsing number`? I'm happy to revert this change for now though if everything else looks good.

Reply:
I guess that ppl search and find #501 (it's the most visited issue) because we mention numeric in many error cases and it's confusing

Had a go at compring jq to `JSON.parse`:
```sh
$ for i in ident truea falsea nana aa1 1aa + - / "&" "*" '(' "'aa'"; do echo "$i:\t$(jq 2>&1 <<< "$i")"; echo "        $(node -p "JSON.parse(\"$i\")" 2>&1 | grep Error | head -n1)"; done
ident:	jq: parse error: Invalid numeric literal at line 2, column 0
        SyntaxError: Unexpected token 'i', "ident" is not valid JSON
truea:	jq: parse error: Invalid literal at line 2, column 0
        SyntaxError: Unexpected non-whitespace character after JSON at position 4 (line 1 column 5)
falsea:	jq: parse error: Invalid literal at line 2, column 0
        SyntaxError: Unexpected non-whitespace character after JSON at position 5 (line 1 column 6)
nana:	jq: parse error: Invalid literal at line 2, column 0
        SyntaxError: Unexpected token 'a', "nana" is not valid JSON
aa1:	jq: parse error: Invalid numeric literal at line 2, column 0
        SyntaxError: Unexpected token 'a', "aa1" is not valid JSON
1aa:	jq: parse error: Invalid numeric literal at line 2, column 0
        SyntaxError: Unexpected non-whitespace character after JSON at position 1 (line 1 column 2)
+:	jq: parse error: Invalid numeric literal at line 2, column 0
        SyntaxError: Unexpected token '+', "+" is not valid JSON
-:	jq: parse error: Invalid numeric literal at line 2, column 0
        SyntaxError: No number after minus sign in JSON at position 1 (line 1 column 2)
/:	jq: parse error: Invalid numeric literal at line 2, column 0
        SyntaxError: Unexpected token '/', "/" is not valid JSON
&:	jq: parse error: Invalid numeric literal at line 2, column 0
        SyntaxError: Unexpected token '&', "&" is not valid JSON
*:	jq: parse error: Invalid numeric literal at line 2, column 0
        SyntaxError: Unexpected token '*', "*" is not valid JSON
(:	jq: parse error: Invalid numeric literal at line 2, column 0
        SyntaxError: Unexpected token '(', "(" is not valid JSON
'aa':	jq: parse error: Invalid numeric literal at line 2, column 0
        SyntaxError: Unexpected token ''', "'aa'" is not valid JSON
```

Reply:
I'd be happy to implement a general `Invalid character "X"` kind of error message, but I may need someone to point me in the right direction. My understanding is that the `check_literal` function returns a `const char*` meaning we wouldn't be able to get that dynamic error message here without changing a bunch of code. If this change is beyond the scope of the issue, I can just revert back to the simple `'` special case.